src/common.cc

```diff
@@ -103,8 +121,9 @@ stim::DetectorErrorModel common::merge_identical_errors(const stim::DetectorErro
     }
   }
   for (const auto& it : errors_by_symptom) {
-    out_dem.append_error_instruction(it.second.probability,
-                                     it.second.symptom.as_dem_instruction_targets(),
+    // Recalculate probability from the final merged cost for the output DEM.
+    double final_p = 1 / (1 + std::exp(it.second.likelihood_cost));
```

Comment:
Nit: it might be nice to have a `get_probability` method (also exposed in pybind) which calculates the probability from the likelihood_cost, which could also be used here.

It's good that we now only store the likelihood cost, but still could be handy to have a getter/setter that uses probability as helper methods.